src/neural/network_st_batch.cc

```diff
@@ -0,0 +1,66 @@
+/*
+  This file is part of Leela Chess Zero.
+  Copyright (C) 2018 The LCZero Authors
+
+  Leela Chess is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  Leela Chess is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with Leela Chess.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "neural/network_st_batch.h"
+
+#include <cassert>
+
+namespace lczero {
+
+SingleThreadBatchingNetwork::SingleThreadBatchingNetwork(
+    std::unique_ptr<Network> parent)
+    : parent_(std::move(parent)) {}
+
+std::unique_ptr<NetworkComputation>
+SingleThreadBatchingNetwork::NewComputation() {
+  ++computations_pending_;
+  return std::make_unique<SingleThreadBatchingNetworkComputation>(this);
+}
+
+void SingleThreadBatchingNetwork::Reset() {
+  assert(computations_pending_ == 0);
+  parent_computation_ = parent_->NewComputation();
+}
+
+SingleThreadBatchingNetworkComputation::SingleThreadBatchingNetworkComputation(
+    SingleThreadBatchingNetwork* network)
+    : network_(network),
+      start_idx_(network_->parent_computation_->GetBatchSize()) {}
+
+void SingleThreadBatchingNetworkComputation::AddInput(InputPlanes&& input) {
+  assert(start_idx_ + batch_size_ ==
+         network_->parent_computation_->GetBatchSize());
+  ++batch_size_;
+  network_->parent_computation_->AddInput(std::move(input));
```

Comment:
It's not rvalue reference but rather universal reference. Whatever that means.
Not sure if std::move() is appropriate there but stackoverflow says I should use std::forward.

Without those it doesn't compile. ("because it's not rvalue reference. If it has name, it's lvalue"). Whatever it means.

I'll surely read more about that. :)

Reply:
On the one hand, I should probably understand this part of the language that this codebase is written in.

On the other hand, fuck C++